src/main/webapp/app/shared/util/firebase/firebase-review-utils.tsx

```diff
@@ -147,29 +146,21 @@ export const getRelevantKeysFromUuidKey = (uuidKey: string) => {
   return relevantKeys;
 };
 
-const getConcatPath = (parentPath: string, currentPath: string) => {
-  return parentPath + FIREBASE_PATH_SEP + currentPath;
-};
-
-const isNestedUnderCreateOrDelete = (parentReview: BaseReviewLevel) => {
-  if (parentReview.reviewLevelType === ReviewLevelType.META) {
-    return parentReview.isUnderCreationOrDeletion;
-  }
-  const parent = parentReview as ReviewLevel;
-  if ([ReviewAction.CREATE, ReviewAction.DELETE, ReviewAction.DEMOTE_MUTATION, ReviewAction.PROMOTE_VUS].includes(parent.reviewAction)) {
-    return true;
-  } else {
-    return parent.isUnderCreationOrDeletion;
+export const joinPathParts = (parentPath: string, ...pathParts: string[]) => {
+  let parts = [];
+  if (parentPath !== '') {
+    parts.push(parentPath);
   }
+  parts = parts.concat(pathParts);
+  return parts.join('/');
 };
 
 const removeLeafNodes = (parentReview: BaseReviewLevel) => {
   for (const key of Object.keys(parentReview.children)) {
     const childReview = parentReview.children[key];
-    if (!childReview.hasChildren()) {
-      if (childReview.reviewLevelType === ReviewLevelType.META && !(childReview as ReviewLevel).review) {
-        delete parentReview.children[key];
-      }
+    if (childReview.hasChildren()) continue;
+    if (childReview.reviewLevelType === ReviewLevelType.META) {
```

Comment:
Looks like this method is only used when you have a grouping review level (META) as a leaf. What scenario would cause this behavior?

Reply:
When looping through an array (mutations, tumors, treatments), I create an initial META review level for each element. The reason is that, I don't know which element has additional reviewable items nested under it. 

If a mutation has no reviewable items, then it has no children, so we can remove the META review level that we created.